refactor(ci): transform conformance snapshot include when output mismatch

overlookmotel opened the pull request:
Include "Output mismatch" in transform conformance snapshots when a test also fails due to scopes/symbols mismatches.

At present, many of the tests are failing on scopes/symbols mismatches, so it's hard to see from snapshots which also have an output mismatch. In particular, when working to fix scope mismatches, it's hard to notice if changes you make cause the output to be wrong. These extra messages make that visible.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5135?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5135** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5135?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#5134** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5134?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Dunqing commented:
Personally, if the output doesn't match we shouldn't show the semantic errors. In

overlookmotel committed:
refactor(ci): transform conformance snapshot include when output mismatch (#5135)

Include "Output mismatch" in transform conformance snapshots when a test also fails due to scopes/symbols mismatches.

At present, many of the tests are failing on scopes/symbols mismatches, so it's hard to see from snapshots which also have an output mismatch. In particular, when working to fix scope mismatches, it's hard to notice if changes you make cause the output to be wrong. These extra messages make that visible.

overlookmotel commented:
@Dunqing Yes maybe that would be a better way to do it. Check that output matches first, and then only check if semantic matches if the output match test passes. You think that'd work?

Dunqing commented:
> @Dunqing Yes maybe that would be a better way to do it. Check that output matches first, and then only check that semantic is correct if the output match test passes. You think that'd work?

Yes, I agree we do that. From my experience development of the transformer. When the output is matched, fixing semantic errors is not too hard
